Document password handling for special characters (#63590)
metabase/metabase: docs/databases/connections/mariadb.md

JSON schema inference doesn't work with MariaDB, due to implementation differences between MySQL and MariaDB.

> ## Password with special characters
> 
> If your password contains character that are non UTF-8, then you might want to add an additional variable to the connection string `passwordCharacterEncoding=<your_encoding_here>`. This will ensure that your MySQL/MariaDB database will understand the special characters of the password when it authenticates.

Reviewer: ```suggestion
## Passwords with special characters

If your password contains characters that aren't UTF-8, then you might need to add an additional variable to the connection string `passwordCharacterEncoding=<your_encoding_here>`. This ensures that MariaDB understands the special characters in the password during authentication.
```
